src/main/java/core/basesyntax/dao/dao/impl/StorageDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.dao.dao.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitModel;
+import java.util.Map;
+
+public class StorageDaoImpl implements StorageDao {
+
+    @Override
+    public String addFruit(FruitModel fruitModel, Integer amount) {
+        for (Map.Entry<FruitModel, Integer> entry: Storage.fruitStorage.entrySet()) {
+            if (entry.getKey().equals(fruitModel)) {
+                int newAmount = entry.getValue() + amount;
+                entry.setValue(newAmount);
+                return fruitModel + " amount was updated: " + Storage.fruitStorage.get(fruitModel);
+            }
+        }
+        Storage.fruitStorage.put(fruitModel, amount);
+        return fruitModel + " added to the storage. Amount: "
+                + Storage.fruitStorage.get(fruitModel);
+    }
+
+    @Override
+    public Integer getFruit(FruitModel fruitModel, int amount) {
```

Comment:
method signature (name + args) is unclear - method is called getFruit, but it accepts amount. Is it something like updateFruit?